Fix setting contrast limits when the histogram widget is open

p-j-smith opened the pull request:
Fixes #272 

- pass a list to `Line2D.set_xdata` rather than a float (`Line2D.set_xdata` [expects an iterable](https://github.com/matplotlib/matplotlib/blob/5cd38c3edcdf0792d0e6aded280a9b7a7de6146f/lib/matplotlib/lines.py#L1290-L1304))
- add test for setting contrast limits when a histogram is plotted

dstansby commented:
Thanks a lot for this long overdue fix! It has been a while since any development happened here, but all the tests seem to be running fine... before merging, could you add a new `2.1.1` section to the changelog (`docs/changelog.rst`) with a short entry for this bugfix?

p-j-smith commented:
Thanks for the speedy review!

I've updated the changelog although I put it under `3.0.1` as the [latest release](https://github.com/matplotlib/napari-matplotlib/releases) is for `3.0.0`.

Also, I've renamed `2.1.0` in the changelog as it looks like `2.1.0` was never released and it went straight to `3.0.0`? Happy to change it back though if this isn't right
